### ISSUE TITLE
Drag Drop in tree access permission issue

### ISSUE THREAD

**Issue by bossloper** [evolution-cms/evolution#472]
A combination of a manager account (not administrator) with security applied to resources does not allow drag and drop on the menu tree to reorder.

- setup a manager user group together with resource group and link them as normal
- create a new manager with the 'Editor' role and make sure in the new user group
- create resources and assign to this resource group. Make sure there are child resources

Log in as the new manager and try to reorder child resources in menu tree.
**Those with the resource group applied cannot be drag sorted.** Even though the manager is in the resource group and can correctly still edit all the resources.
Those with no permission group (public) can be dragged.
![drag](https://user-images.githubusercontent.com/1744687/35453244-60619f5a-02c1-11e8-816e-1a27f8e5c790.png)

**Comment by 64j** [comment]
@bossloper 
Dragging resources works only for the Administrator role.
To avoid "curved hands" managers

**Comment by 64j** [comment]
@bossloper 
And do not even insist to allow managers to drag resources.
Either administrators, or none!

**Comment by bossloper** [comment]
But a manager (not admin) can drag resources when there is no permission on the resource i.e. it is set to 'All Resource Groups (Public)'

**Comment by 64j** [comment]
https://github.com/evolution-cms/evolution/blob/develop/manager/media/style/default/ajax.php#L486

Yes.
This should be corrected.

Do I understand correctly?
It is necessary only to administrators, to allow drag and drop?

**Comment by bossloper** [comment]
RE: It is necessary only to administrators, to allow drag and drop?
- I have some sites where the manager is very good so would be nice if there was a system configuration to enable/disable tree dragging for none administrators.... or maybe a new manager role would work well?

**Comment by 64j** [comment]
@bossloper 
So if the manager is so good, you need to do his administrator)
Believe me, most managers with their hands in the a**.

**Comment by bossloper** [comment]
... not that good. :-)I don't want anyone else near the elements/roles/configuration..

**Comment by Nicola1971** [comment]
Agree with @bossloper.
I don't use it, but my clients really need and asked for years.
It's really nice feature for editors and publishers, because is for who create content.

And is for who don't care about 404 errors 😄 

For me should be for any user allowed to create and edit documents.

Same for global tabs, because they need to copy and paste content from other resources.

**Comment by pmfx** [comment]
+1 it would be great to have tree drag&drop and global tabs available as user/role permissions.

**Comment by Nicola1971** [comment]
@64j  because I have not understand all your post, for the language barrier  :)   is there a technical problem to allow users to drag resources or use global tabs?

**Comment by 64j** [comment]
@Nicola1971 
There are no technical problems.
It is necessary to prohibit dragging for managers so that they do not accidentally spoil it with a single click of the mouse

**Comment by Nicola1971** [comment]
@64j so is better allow developer/webmaster choose who can use this feature, beacause is not critical for all kind of users, but I agree - drag and drop for some users can be very dangerous.

**Comment by bossloper** [comment]
Would it be possible to add a confirmation dialog when dropping the drag? It would help prevent accidents.

**Comment by 64j** [comment]
So there were those who have full rights.
And in consequence of this broke the site.

**Comment by fourroses666** [comment]
Maybe we can add some small icons on the right side of the tree; like the subnav popup and the drag-and-drop icon? Both are not directly visible options. 

And maybe confirmation drag-and-drop for non admins?

**Comment by 64j** [comment]
Yes it can be added, but we have a document tree very small that I'm afraid there will not be enough space.
It is necessary to think, and choose a solution

**Comment by risingisland** [comment]
Is there some place that this can be disabled? 
I looked in roles, but did not see anything.

**Comment by pmfx** [comment]
We really need to add a new user role permissions for drag&drop and global tabs features.
